### ISSUE TITLE
[HOLD for payment 2023-08-28] [$1000] [Manual Requests] Refactor usages of action.originalMessage to the transactions key in Onyx

### ISSUE THREAD

**Issue by luacmartins** [Expensify/App#22208]
# Problem
We are now storing the transactions key in Onyx and should no longer use `action.originalMessage.<amount|comment|date|etc>`.

# Why is this important?
Avoids data duplication and keeps the codebase clean and less confusing

# Solution
Refactor the code!
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~014f38a35df644bdf5</li>
        <li>Upwork Job ID: 1676296881515593728</li>
        <li>Last Price Increase: 2023-07-05</li>
    </ul>
</details>

**Comment by luacmartins** [comment]
This is purely a code refactor and doesn't touch any design. So I'm gonna unassign @shawnborton.

**Comment by kowczarz** [comment]
Hello! I'm from Software Mansion, I'd like to work on this job.

**Comment by kowczarz** [comment]
Am I correct and the scope of the task is only `src/libs/actions/Report.js` or should I do it in all actions? And where can I find more docs about the new convention?

**Comment by luacmartins** [comment]
Hey @kowczarz. No, the scope is the entire App. We want to eventually deprecate originalMessage.

**Comment by luacmartins** [comment]
What docs are you looking for? The data currently found in originalMessage should be available in the new transactions key

**Comment by kowczarz** [comment]
Hey I had some PR in review and I had to introduce some changes to them, but I think since now I can focus full time on this task. I will try to prepare a proposal in the following days.

**Comment by luacmartins** [comment]
Sounds good! Let me know if you need any help!

**Comment by luacmartins** [comment]
Still waiting on a proposal from @kowczarz

**Comment by luacmartins** [comment]
@kowczarz reached out 1:1 to ask about API not returning the `transactions_` key. @kowczarz did you manage to fix the issue after my latest reply?

**Comment by luacmartins** [comment]
@kowczarz any updates?

**Comment by adelekennedy** [comment]
pending proposal from @kowczarz

**Comment by staszekscp** [comment]
Olá @luacmartins! Because @kowczarz has some other tasks he's been working on ([including one that is pretty big](https://github.com/Expensify/App/issues/16805)), he'll hand it over another developer that would be able to focus only on this task 😄 Shortly we will let you know who it is going to be, so we can start moving forward! Therefore you may expect someone posting a comment to pick it up 😄

**Comment by luacmartins** [comment]
@Nikhil-Vats thanks for the interest! It seems like we're reassigning this already. Thanks for the heads up @staszekscp, looking forward to the comment.

**Comment by sebryu** [comment]
Hey @luacmartins  👋 I'm from Software Mansion, and I will take it over after kowczarz

**Comment by luacmartins** [comment]
Latest update [here](https://expensify.slack.com/archives/C04878MDF34/p1690817885663589)

**Comment by luacmartins** [comment]
@sebryu would you mind providing a high level plan for this issue?

**Comment by sebryu** [comment]
@luacmartins the high level plan is to thoroughly go through usages of `originalMessage` and see which of the field is duplicated with some other available field either in the same `action` or in related key in `transactions_`. Part of this job is already done, I just need some more time to finish that and create a proposal.

**Comment by luacmartins** [comment]
Sounds good and thanks for the update! When do you think you'd have the proposal ready? No rush, just trying to gauge when we can expect this.

**Comment by sebryu** [comment]
Tomorrow proposal should be ready

**Comment by sebryu** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
We have duplicated data in action.originalMessage.
### What is the root cause of that problem?
Backend now will return some fields from action.originalMessage under new transactions key.
### What changes do you think we should make in order to solve the problem?

After reviewing code I see that we should replace usages of some originalMessage fields to the new keys. In some cases we would take relatedTransaction in the following way: 
```
const transaction = allTransactions[`${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`];
``` 
<table>
<tr>
<th>Old </th>
<th>New </th>
</tr>
<tr>
<td>
action.originalMessage.amount
</td>
<td>
relatedTransaction.amount
</td>
</tr>
<tr>
<td>
action.originalMessage.currency
</td>
<td>
relatedTransaction.currency
</td>
</tr>
<tr>
<td>
action.originalMessage.comment
</td>
<td>
relatedTransaction.comment.comment
</td>
</tr>
<tr>
<td>
reportAction.originalMessage.linkedReportID
</td>
<td>
action.childReportId (but this field is number so additional `.toString() needed)
</td>
</tr>
<tr>
<td>
reportAction.originalMessage.lastModified
</td>
<td>
action.lastModified
</td>
</tr>
</table>

Example reportAction.originalMessage:
```
"originalMessage": {
    "IOUReportID": 1124914273603956,
    "IOUTransactionID": "7830425852445078818",
    "amount": -333,
    "comment": "first split",
    "currency": "PLN",
    "lastModified": "2023-07-31 15:16:12.319",
    "participantAccountIDs": [
        15159786,
        15159790
    ],
    "participants": [
        "sebastian.szewczyk+1@swmansion.com",
        "sebastian.szewczyk+2@swmansion.com"
    ],
    "type": "create"
},
````
Related transaction:
```
"transactions_7830425852445078818": {
    "amount": -333,
    "comment": {
        "comment": "first split",
        "originalTransactionID": "4617747673208542386",
        "source": "split"
    },
    "created": "2023-07-31 15:16:12",
    "currency": "PLN",
    "merchant": "IOU",
    "modifiedAmount": 0,
    "modifiedCreated": "",
    "modifiedCurrency": "",
    "modifiedMerchant": "",
    "transactionID": "7830425852445078818"
}
```

Other originalMessage fields which are NOT migrated (in various onyx actions), but they are used in the code:
- IOUTransactionID
- IOUReportID
- participantAccountIDs
- participants
- type
- reason
- newAccountID
- oldAccountID
- events
- oldName
- newName
- IOUDetails
- source (this one seems deprecated already)

Please do let me know if any of the above fields should also be changed.

### What alternative solutions did you explore? (Optional)

**Comment by luacmartins** [comment]
Proposal looks good @sebryu! My only suggestion is to drop the `related` prefix and go with just `transaction.<key>`, other than that, we should be good to start implementing this.

**Comment by mountiny** [comment]
@sebryu Looking good, lets prioritise this PR now so we can get it merged ASAP. We are working on feature which allows editing money requests and this is required to make it work correctly

**Comment by luacmartins** [comment]
@sebryu we discussed internally and since this issue is blocking one of our top initiatives, I'll be taking it over. Thanks for the time you spent investigating this and let me know if you have any questions.

**Comment by sebryu** [comment]
Oh, I see. Thanks for the heads up, and sorry that I did not manage to make it in your expected timeline 🙇

**Comment by luacmartins** [comment]
We're over the regression period. I see that nobody requires payment here, so I'm gonna close this issue. Please let me know if I missed something and we need to reopen this issue.

>@sebryu does not require payment (Contractor)
@sobitneupane does not require payment (Eligible for Manual Requests)

**Comment by mountiny** [comment]
@sobitneupane @adelekennedy I think we still need to post the summarizing post for what the payments for Manual requests will be

**Comment by sobitneupane** [comment]
Requested payment on newDot

https://github.com/Expensify/App/issues/22208#issuecomment-1701555382

**Comment by JmillsExpensify** [comment]
$500 approved for payment via NewDot based on BZ summary.
